File: src/main/java/com/github/pauldambra/moduluschecker/ModulusChecker.java
package com.github.pauldambra.moduluschecker;

import com.github.pauldambra.moduluschecker.Account.BankAccount;
import com.github.pauldambra.moduluschecker.valacdosFile.ModulusWeightRows;

import java.io.IOException;

public class ModulusChecker {
    private SortCodeSubstitution sortCodeSubstitution = SortCodeSubstitution.fromFile("file/scsubtab.txt");

    private ModulusWeightRows weightRows = ModulusWeightRows.fromFile("file/valacdos.txt");

    public ModulusChecker() throws IOException {
    }

    public Boolean checkBankAccount(String sortCode, String accountNumber) {
        final BankAccount account = BankAccount.Of(sortCode, accountNumber);
        final ModulusCheckParams params = ModulusCheckParams.startingParams(account);

        return ModulusCheckingChain
                .create(weightRows, sortCodeSubstitution)
                .check(params)
                .processResults();
    }

}

File: src/main/java/com/github/pauldambra/moduluschecker/ModulusChecker.kt
package com.github.pauldambra.moduluschecker

import com.github.pauldambra.moduluschecker.account.BankAccount
import com.github.pauldambra.moduluschecker.valacdosFile.ModulusWeightRows

import java.io.IOException

class ModulusChecker @Throws(IOException::class)
constructor() {
    private val sortCodeSubstitution = SortCodeSubstitution.fromFile("file/scsubtab.txt")

    private val weightRows = ModulusWeightRows.fromFile("file/valacdos.txt")

    fun checkBankAccount(sortCode: String, accountNumber: String): Boolean {
        val account = BankAccount(sortCode, accountNumber)
        val params = ModulusCheckParams(account)

        return ModulusCheckingChain
          .create(weightRows, sortCodeSubstitution)
          .check(params)
          .processResults()
    }

}
